### DISCUSSION TITLE
Bug in getting week of the year

### DISCUSSION THREAD

**Discussion by aveTakashi** [moment/moment#6427] [Q&A] [answered]
**Describe the bug**
It seem that if  a particular week ends and starts a year the return week number is wrong for the new year

**To Reproduce**
```
"Correct Week: " + moment("2020-12-28").isoWeek(); //returns 53

"Wrong Week: " + moment("2021-01-01").isoWeek(); //returns 53

```
**Expected behavior**
I expect the first line of code to return 53 (counting from 0) and the second line of code to return 0 as the first week

**Screenshots**
[https://plnkr.co/plunk/iBLE6W3p3HTWCMkm](url)

**Comment by AdiechaHK** [comment]
@aveTakashi , I was checking out the issue and looking for the solution, meanwhile, I found the test cases, which show those are expected to be considered in the last week of the previous year.

ref: https://github.com/moment/moment/blob/develop/src/test/moment/weeks.js#L93-L102

**Comment by ashsearle** [comment, marked as answer]
ISO weeks go from 1 to 53. The week runs from Monday to Sunday, and the first week 1 of the year always contains 4th Jan.

**Comment by tehdb** [comment]
[According to Wikipedia](https://en.wikipedia.org/wiki/ISO_week_date): An ISO week-numbering year (also called ISO year informally) has 52 or 53 full weeks. That is 364 or 371 days instead of the usual 365 or 366 days. These 53 week years occur on all years that have Thursday as the 1st of January and on leap years that start on Wednesday the 1st.

**Comment by commuze1** [comment]
+1
